### PR TITLE
ci: Add a script to generate changes to ci-config

### PR DESCRIPTION
**Pull request by JST5000**
## High Level Overview of Change

Automate figuring out which amendments to enable in our CI docker container.

### Context of Change

Periodically we need to update the `rippled.cfg` we use in the standalone node we use during CI so that the library is being tested against the latest amendments. This adds a script to get the hashes of any new amendments so it's easy to update our CI without a bunch of manual lookups / comparisons.

### Type of Change

- [X] New feature (non-breaking change which adds functionality)
- [X] Documentation Updates

## Test Plan

Manually tested the script against various networks / removing entries from the config file to see if it noticed.

## Future Tasks
Automatically generate a PR to update the CI file directly with hashes + names instead of requiring a manual lookup on the known amendments page.

### CODE REVIEW

File: .ci-config/rippled.cfg
Comment (ckniffen) [inline .ci-config/rippled.cfg:167]: CryptoConditionsSuite is super old deprecated. https://xrpscan.com/amendment/86E83A7D2ECE3AD5FA87AB2195AE015C950469ABF0B72EAACED318F74886AE90

File: .ci-config/rippled.cfg
Comment (ckniffen) [inline .ci-config/rippled.cfg:169]: Should probably be added above since it was much older. Added in 1.9.1.

File: .ci-config/rippled.cfg
Comment (ckniffen) [inline .ci-config/rippled.cfg:167]: Move between 1.10 and 1.12.  It was added in 1.11.0.

File: .ci-config/rippled.cfg
Comment (ckniffen) [reply .ci-config/rippled.cfg:169]: Might need to add an integration test for `WalletLocator` and the size of SignerLists

File: .ci-config/rippled.cfg
Comment (JST5000) [reply .ci-config/rippled.cfg:167]: It's still enabled on mainnet though

File: .ci-config/getNewAmendments.js
Comment (ckniffen) [inline .ci-config/getNewAmendments.js:21]: Nit: you can import the constant from `xrpl`.

File: packages/ripple-binary-codec/HISTORY.md
Comment (ckniffen) [inline packages/ripple-binary-codec/HISTORY.md:6]: nit: note that this includes AMM definitions updates too.

File: .ci-config/getNewAmendments.js
Comment (JST5000) [reply .ci-config/getNewAmendments.js:21]: I don't think it's properly exported outside the repo. Importing it directly would require compiling the typescript, which would make it more annoying to run the script, so I don't think it's worth doing that for this.

File: packages/ripple-binary-codec/HISTORY.md
Comment (mvadari) [reply packages/ripple-binary-codec/HISTORY.md]: Shouldn't the AMM definitions wait for the AMM PR instead of being done here?

**Review by mvadari — COMMENTED** [review]
Can this automatically be run in a Github Action, say once a week?

File: packages/ripple-binary-codec/HISTORY.md
Comment (JST5000) [reply packages/ripple-binary-codec/HISTORY.md]: I'd prefer to just match rippled's `develop` as best as possible instead of manually editing the `definitions.json` file. Plus, I don't think there's any harm in having more `definitions` available.

Comment (mvadari) [reply packages/ripple-binary-codec/HISTORY.md]: Can we hold off on this PR until that's merged in, then? Since that one's going to be merged soon anyways.

Comment (JST5000) [reply packages/ripple-binary-codec/HISTORY.md]: This was generated from `develop` already - not sure why the definitions were there before the big AMM PR, but it seems they're already defined.

File: .ci-config/rippled.cfg
Comment (JST5000) [reply .ci-config/rippled.cfg]: Would it be alright to leave that for a separate ticket? (I'm not exactly sure what we'd be testing for that - maybe you can flesh out an Issue to add it w/ an idea for what we'd want to test on the client library side?)

File: .ci-config/getNewAmendments.js
Comment (ckniffen) [reply .ci-config/getNewAmendments.js:21]: Will need to look into how why it is not on the exported object.

**Review by ckniffen — APPROVED** [review]

**Review by jonathanlei — APPROVED** [review]

**Comment by mvadari** [comment]
super nit: could the PR title avoid the word "automate"? That's part of the reason why I asked about a GitHub Action - I was expecting something that automatically ran the script.

**Comment by JST5000** [comment]
> super nit: could the PR title avoid the word "automate"? That's part of the reason why I asked about a GitHub Action - I was expecting something that automatically ran the script.

I think I wanted to automate it originally, but didn't see an easy way - I think your question was fair, I just didn't find an answer just yet and might look into it more as a follow up task. Updated the title for now since I think the changes are helpful as is.

**Pull request merged**
